src/arithmetic/string_funcs/string_funcs.c

```diff
@@ -72,20 +72,30 @@ SIValue AR_RTRIM(SIValue *argv, int argc) {
 	return SI_TransferStringVal(trimmed);
 }
 
-/* returns a string in which the order of all characters in the original string have been reversed. */
+/* incase the parameter type is 
+ * 1. string - returns a string in which the order of all characters in the original string have been reversed.
+ * 1. array  - returns an array in which the order of all elements in the original array have been reversed. */
 SIValue AR_REVERSE(SIValue *argv, int argc) {
 	if(SIValue_IsNull(argv[0])) return SI_NullVal();
-	char *str = argv[0].stringval;
-	size_t str_len = strlen(str);
-	char *reverse = rm_malloc((str_len + 1) * sizeof(char));
 
-	int i = str_len - 1;
-	int j = 0;
-	while(i >= 0) {
-		reverse[j++] = str[i--];
+	SIValue value = argv[0];
+	if(SI_TYPE(value) == T_STRING) {
+		char *str = value.stringval;
+		size_t str_len = strlen(str);
+		char *reverse = rm_malloc((str_len + 1) * sizeof(char));
+
+		int i = str_len - 1;
+		int j = 0;
+		while(i >= 0) {
+			reverse[j++] = str[i--];
+		}
```

Comment:
another option would be duplicate `src` to `reverse` and in-place reverse `reverse` the loop would perform n/2 iterations

Reply:
the loop will be n/2 iteration but the body will do 3 operations for switch the value